lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -0,0 +1,962 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- |
+-- Copyright: © 2020 IOHK
+-- License: Apache-2.0
+--
+-- Network Layer for talking to Haskell re-written nodes.
+--
+-- Good to read before / additional resources:
+--
+-- - Module's documentation in `ouroboros-network/typed-protocols/src/Network/TypedProtocols.hs`
+-- - Data Diffusion and Peer Networking in Shelley (see: https://raw.githubusercontent.com/wiki/input-output-hk/cardano-wallet/data_diffusion_and_peer_networking_in_shelley.pdf)
+--     - In particular sections 4.1, 4.2, 4.6 and 4.8
+module Cardano.Wallet.Shelley.Network
```

Comment:
Make sure to base this "fork" on the latest version of the Byron Network code which includes Rodney's latest refactoring and inclusion of the local state query (you can use a "null client" here if the protocol isn't available yet on the node's end, but at least, we have the skeleton in place on the wallet's side).

Reply:
> if the protocol isn't available yet on the node's end

It was, was I was grepping wrong.